Build-Dependencies.ps1

```diff
@@ -62,7 +63,15 @@ function Run-Stages {
 
         . $Dependency
 
-        if ( ! ( $Targets.contains($Target) ) ) { continue }
+        if ( ! ( $Targets.contains($Target) ) ) {
+            $Stages | ForEach-Object {
+                Log-Debug "Removing function $_"
+                Remove-Item -ErrorAction 'SilentlyContinue' function:$_
+                $script:StageName = ''
+            }
+
+            return
+        }
```

Comment:
What is the intent here?

As an aside, although this `if` is not new, it seems like the `if` test will always return `! ( $true )`, since `$Targets` is set to `@($script:Target)` just above. Perhaps I've misunderstood something.

Reply:
`continue` doesn't work as it would in a `foreach` blog - instead it continues the execution of the current script block and breaks the entire script.

`ForEach-Object` is literally borrowing behaviour from functional languages, in which the script block is an anonymous script block. Skipping an element thus requires simply returning from the current block early.

`$Stages` exists outside of that scope and is stateful, so an early return requires cleaning up all stage functions created by the dot import of the script file. Without this, the `Patch` function of an earlier dependency would be executed over and over again until a later dependency overwrites it with its own `Patch` step.

$Targets is overwritten with every dot import of a new dependency script and needs to be evaluated on every run of the anonymous script block.

Reply:
Discussed further off-thread.